plugins/tiddlywiki/webdav/webdavadaptor.js

```diff
@@ -0,0 +1,248 @@
+				for(var i=1; i< items.length; i++) {
+					var response	 = items[i];
+					var href		 = response.getElementsByTagNameNS(ns, 'href')[0].firstChild.nodeValue;
+					href = href.replace(/\/$/, ''); // Strip trailing slash
+					href = href.replace(self.tiddlersBaseURL + "/", ""); // Strip path
+					href = decodeURIComponent(href);
+					var propstat	 = response.getElementsByTagNameNS(ns, 'propstat')[0];
+					var prop		 = propstat.getElementsByTagNameNS(ns, 'prop')[0];
+					var resourcetype = prop.getElementsByTagNameNS(ns, 'resourcetype')[0];
+					var collection   = resourcetype.getElementsByTagNameNS(ns, 'collection')[0];
+
+					if (prop.getElementsByTagNameNS(ns, 'getcontenttype').length > 0) {
+						// var type = prop.getElementsByTagNameNS(ns, 'getcontenttype')[0].innerHTML;
+						if (href.endsWith(".tid") && href!=="$__StoryList.tid") {
+							var type = "application/x-tiddler";
+							// var revision	 = prop.getElementsByTagNameNS(ns, 'getetag')[0].innerHTML;
+							var revision	 = prop.getElementsByTagNameNS(ns, 'getlastmodified')[0].innerHTML;
+						  href = href.slice(0, -4);
+						  tiddlers.push({ title: href,
+										revision: revision,
+										type: type });
+						}
+					}
+					if (collection) {
+						if (!(self.knownDirs.includes(href))) {
+							self.knownDirs.push(href);
+						}
+					}
+				}
+				callback(null, tiddlers);
+			}
+		}
+	});
+};
+
+/*
+Save a tiddler and invoke the callback with (err,adaptorInfo,revision)
+*/
+WebDAVAdaptor.prototype.saveTiddler = function(tiddler,callback) {
+	var self = this;
+	// Save the tiddler as a self contained templated file
+	var content = self.wiki.renderTiddler("text/plain","$:/core/templates/tid-tiddler",
+		{variables: {currentTiddler: tiddler.fields.title}});
+	var url = this.tiddlersBaseURL + "/" + encodeURIComponent(tiddler.fields.title) + ".tid";
+	var path = dirname(tiddler.fields.title);
+	if (path!==tiddler.fields.title && !(self.knownDirs.includes(path))) {
+		this.mkdir_p(path, callback);
+	}
+	$tw.utils.httpRequest({
+		url: this.tiddlersBaseURL + "/" + tiddler.fields.title + ".tid",
+		type: "PUT",
+		headers: {
+			"Content-type": "text/plain"
+		},
+		data: content,
+		callback: function(err,data,request) {
+			if(err) {
+				return callback(err);
+			}
+			// Save the details of the new revision of the tiddler
+			var revision = new Date().toGMTString(); // request.getResponseHeader("Etag");
+
+			// Invoke the callback
+			callback(null, {}, revision);
+		}
+	});
+};
+
+/*
+Load a tiddler and invoke the callback with (err,tiddlerFields)
+*/
+WebDAVAdaptor.prototype.loadTiddler = function(title,callback) {
+	var self = this;
+	$tw.utils.httpRequest({
+		url: this.tiddlersBaseURL + "/" + title + ".tid",
+		headers: {accept: "charset=UTF-8"},
+		callback: function(err,data,request) {
+			if(err) {
+				return callback(err);
+			}
+			// Invoke the callback
+			var type = "application/x-tiddler";
```

Comment:
tiddlers are type: `text/vnd.tiddlywiki`, `application/x-tiddler` is tiddlywiki-classic

Reply:
Um, I think that's the other way around. `application/x-tiddler` is TiddlyWiki 5 if I'm reading boot.js correctly (line 1300).